sympy/integrals/transforms.py

```diff
@@ -1937,6 +1937,98 @@ def laplace_transform(f, t, s, legacy_matrix=True, **hints):
 
     return LaplaceTransform(f, t, s).doit(**hints)
 
+def _facorize_num_den(f, s):
+    from sympy.simplify.radsimp import fraction
+    [n, d] = fraction(f)
+    try:
+        nf = n.factor(s)
+    except PolynomialError:
+        nf = n
```

Comment:
What is an example of `n` and `s` such that this would raise `PolynomialError`?

Reply:
It could contain non-polinomial factors like `exp(-a*s)`. Would it be better to use `.is_polynomial()` on `n` and `d`?

Reply:
Why use `n.factor(s)` rather than just `n.factor()` or `factor(n)`:
```python
In [6]: e = expand((s**2 - exp(s))**2)

In [7]: e
Out[7]: 
 4      2  s    2⋅s
s  - 2⋅s ⋅ℯ  + ℯ   

In [8]: factor(e)
Out[8]: 
         2
⎛ 2    s⎞ 
⎝s  - ℯ ⎠ 

In [9]: factor(e, s)
---------------------------------------------------------------------------
PolynomialError    
```

Reply:
Does this whole function do something better than just using `factor`?

Reply:
It's purpose is to use `factor(s)` on numerator and denominator independently to prepare the function in a way that facilitates the pattern match with functions containing products of first- and second-order polynomials in `s`.

This is also why just using `factor()` won't be useful here.

It may be the case that factoring the numerator is unnecessary if the new function applying rules is only used as it is now, but I think that might change later in the development process: 3rd- and 4th-order expression may require it.

Reply:
The ordinary `factor` function factorises the numerator and denominator:
```python
In [3]: f = expand((x - 1)*(x - 2))/expand((x - 3)*(x - 4))

In [4]: f
Out[4]: 
  2          
 x  - 3⋅x + 2
─────────────
 2           
x  - 7⋅x + 12

In [5]: factor(f)
Out[5]: 
(x - 2)⋅(x - 1)
───────────────
(x - 4)⋅(x - 3)

In [6]: n, d = fraction(f)

In [7]: factor(n, x) / factor(d, x)
Out[7]: 
(x - 2)⋅(x - 1)
───────────────
(x - 4)⋅(x - 3)
```
Can you give an example to show that this function is different from `factor`?

Reply:
No. I seem to have tried `factor` with something that it cannot factor.